conans/cli/commands/list.py

```diff
@@ -134,68 +163,76 @@ def _get_remotes(conan_api, args):
 @conan_subcommand(formatters=list_recipe_revisions_formatters)
 def list_recipe_revisions(conan_api, parser, subparser, *args):
     """
-    List all the revisions of a recipe
+    List all the revisions of a recipe reference.
     """
     _add_common_list_subcommands(subparser, "reference")
     args = parser.parse_args(*args)
 
-    if not args.cache and not args.remote and not args.all_remotes:
-        # If neither remote nor cache are defined, show results only from cache
-        args.cache = True
-
-    remotes = _get_remotes(conan_api, args)
-
+    use_remotes = any([args.remote, args.all_remotes])
     results = []
-    if args.cache:
-        result = {
-            'remote': None,
-            'reference': args.reference,
-            'results': conan_api.get_recipe_revisions(args.reference)
-        }
-        results.append(result)
 
-    for remote in remotes:
-        result = {
-            'remote': remote.name,
-            'reference': args.reference,
-            'results': conan_api.get_recipe_revisions(args.reference, remote=remote)
-        }
+    # If neither remote nor cache are defined, show results only from cache
+    if args.cache or not use_remotes:
+        result = conan_api.get_recipe_revisions(args.reference)
         results.append(result)
 
+    if use_remotes:
+        remotes = _get_remotes(conan_api, args)
+        for remote in remotes:
+            result = conan_api.get_recipe_revisions(args.reference, remote=remote)
+            results.append(result)
+
     return results
 
 
 @conan_subcommand(formatters=list_package_revisions_formatters)
 def list_package_revisions(conan_api, parser, subparser, *args):
     """
-    List all the revisions of a package ID
+    List all the revisions of a package ID reference.
     """
     _add_common_list_subcommands(subparser, "package_reference")
     args = parser.parse_args(*args)
 
-    if not args.cache and not args.remote and not args.all_remotes:
-        # If neither remote nor cache are defined, show results only from cache
-        args.cache = True
+    use_remotes = any([args.remote, args.all_remotes])
+    results = []
+
+    # If neither remote nor cache are defined, show results only from cache
+    if args.cache or not use_remotes:
+        result = conan_api.get_package_revisions(args.package_reference)
+        results.append(result)
+
+    if use_remotes:
+        remotes = _get_remotes(conan_api, args)
+        for remote in remotes:
+            result = conan_api.get_package_revisions(args.package_reference, remote=remote)
+            results.append(result)
+
+    return results
+
 
-    remotes = _get_remotes(conan_api, args)
+@conan_subcommand(formatters=list_package_ids_formatters)
+def list_package_ids(conan_api, parser, subparser, *args):
+    """
+    List all the package IDs for a given recipe reference. If it doesn't include the revision,
+    the command will retrieve all the package IDs for the latest one by default.
```

Comment:
A little confusing. I'd rephrase it slightly:

List all the package IDs for a given recipe reference. If the reference doesn't include the recipe revision, the command will retrieve all the package IDs for the most recent revision.